libmproxy/protocol/http.py

```diff
@@ -1457,9 +1457,11 @@ def run(self):
                     server = ServerConnection(server_address)
                     server.connect()
                     if r.scheme == "https":
-                        server.establish_ssl(self.config.clientcerts, sni=self.flow.server_conn.sni)
+                        sni = None
+                        if self.flow.server_conn:
+                            sni = self.flow.server_conn.sni
+                        server.establish_ssl(self.config.clientcerts, sni=sni)
```

Comment:
These changes shouldn't be neccessary. We should always assume there's a ServerConnection.